packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1657,6 +1657,19 @@ class TextSelectionGestureDetectorBuilder {
         && renderEditable.selection!.end >= textPosition.offset;
   }
 
+  bool _tapWasOnSelection(Offset position) {
+    if (renderEditable.selection == null) {
+      return false;
+    }
+
+    final TextPosition textPosition = renderEditable.getPositionForPoint(
+      position,
+    );
+
+    return renderEditable.selection!.start < textPosition.offset
+        && renderEditable.selection!.end > textPosition.offset;
```

Comment:
Will this work if the selection is collapsed?

Reply:
Talked offline about not handling `==`.

The reason I didn't handle the `==` case is because say we have `hello world` with `world` selected, and we want to tap at the edge of `world` to collapse the selection and move the cursor to the end of the field. Because we handle `==` we can't do that and the selection stays on `world`.